testapi.pm

```diff
@@ -275,7 +275,15 @@ sub _check_backend_response {
             }
         }
         if (!$check && !$rsp->{saveresult}) {
-            OpenQA::Exception::FailedNeedle->throw(error => "no candidate needle with tag(s) '$mustmatch' matched", tags => $mustmatch);
+            # Must match can be only scalar or array ref.
+            my $needletags = $mustmatch;
+            if (ref($mustmatch)) {
```

Comment:
Thanks for the input.

That piece of code is either invoked by the check_screen or assert_screen. AFAIK, the tests are using only scalars and arrayrefs.